lib/array_equals.rb

```diff
@@ -1,5 +1,24 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+
+  i= 0
+  if(array1 == nil) && (array2 == nil)
+    return true
+  elsif (array1 == nil) && (array2 != nil)
+    return false
+  elsif (array1 != nil) && (array2 == nil)
+    return false
+  elsif array1.length != array2.length
+    return false
+  else
+    array1.length.times do
+      if array1[i] != array2[i]
+        return false
+      else
+        i += 1
+      end
+    end
+  end
```

Comment:
Since all the `if-elsif` statements return, you can just make this loop outside the if-else block.

```suggestion
  array1.length.times do
    if array1[i] != array2[i]
      return false
    else
      i += 1
    end
  end
```